File: Assessment/src/pages/CartPage.py
class CartPage:

    def __init__(self, page):
        self.tbl_cart = page.wait_for_selector("table > tbody")
        self.row_tbl = self.tbl_cart.query_selector_all('tr')
        self.total_cell = page.locator("tfoot > tr:nth-child(1) > td > strong").inner_text()

    # get items
    def get_cart_details(self):
        item_1 = []
        item_2 = []
        item_3 = []
        row_count = 0
        quantity = any
        for row in self.row_tbl:
            cells = row.query_selector_all('td')
            for cell in cells:
                cell_val = cell.text_content()
                quantity = cell.query_selector_all('input')
                if row_count == 0:
                    if len(quantity) == 0:
                        if '$' in cell_val:
                            split_val = cell_val.split(sep='$')
                            item_1.append(split_val[1])
                        else:
                            item_1.append(cell_val.lstrip())
                    else:
                        for qty in quantity:
                            item_1.append(qty.input_value())
                elif row_count == 1:
                    if len(quantity) == 0:
                        if '$' in cell_val:
                            split_val = cell_val.split(sep='$')
                            item_2.append(split_val[1])
                        else:
                            item_2.append(cell_val.lstrip())
                    else:
                        for qty in quantity:
                            item_2.append(qty.input_value())
                elif row_count == 2:
                    if len(quantity) == 0:
                        if '$' in cell_val:
                            split_val = cell_val.split(sep='$')
                            item_3.append(split_val[1])
                        else:
                            item_3.append(cell_val.lstrip())
                    else:
                        for qty in quantity:
                            item_3.append(qty.input_value())
            row_count += 1
        print(item_1, item_2, item_3)
        return item_1, item_2, item_3

    @property
    def get_total(self):
        total = self.total_cell.split()
        return float(total[1])

File: Assessment/src/pages/CartPage.ts

import { expect, Page } from '@playwright/test'

export default class CartPage { 
    constructor(private page: Page) {}

    tbl_cart = '//div[2]/div/form/table/tbody'
    row_count = 'tr'
    total_cell = '//tfoot/tr/td/strong'

    // get items
    async get_cart_details() {
        let item_1 = []
        let item_2 = []
        let item_3 = []
    
        await expect(this.page.locator(this.tbl_cart)).toBeVisible();
        let row_tbl = this.page.locator(this.tbl_cart).locator(this.row_count);
        const row_ctr = await row_tbl.count();

        for (let i = 0; i < row_ctr; i++) {
            const row = row_tbl.nth(i);
            const col = row.locator('td');
            const col_ctr = await col.count();
            for (let j = 0; j < col_ctr; j++) {
                let cell = await col.nth(j).innerText();
                let quantity = col.nth(j).locator('input');  
                let qty_ctr = await quantity.count();
                let qty_val;
                if (i == 0) {
                    if (qty_ctr == 0) {
                        if (cell.includes('$')) {
                            let split_val = cell.toString().split('$')
                            item_1.push(split_val[1])
                        }
                        else
                            item_1.push(cell.toString().trimStart())
                    }
                    else
                        qty_val = await quantity.inputValue();
                        item_1.push(qty_val)
                    
                }
                else if (i == 1) {
                    if (qty_ctr == 0) {
                        if (cell.includes('$')) {
                            let split_val = cell.toString().split('$')
                            item_2.push(split_val[1])
                        }
                        else
                            item_2.push(cell.toString().trimStart())
                    }
                    else
                        qty_val = await quantity.inputValue();
                        item_2.push(qty_val)
                }
                else if (i == 2) {
                    if (qty_ctr == 0) {
                        if (cell.includes('$')) {
                            let split_val = cell.toString().split('$')
                            item_3.push(split_val[1])
                        }
                        else
                            item_3.push(cell.toString().trimStart())
                    }
                    else
                        qty_val = await quantity.inputValue();
                        item_3.push(qty_val)
                }
            }
        }
        return { item_1, item_2, item_3 };
    }

    // get total amount
    async get_total() : Promise<number> {
        let total_row = await this.page.locator(this.total_cell).innerText();
        let total = total_row.toString().split(' ')
        return Number(total[1])
    }

}